tests/pytests/test_defrag.py

```diff
@@ -0,0 +1,233 @@
+            (if None use temporary directory at @path)
+
+    Yields:
+        Servers: configured server with one backend and specified paramters.
+    """
+    config = [{
+        'backends': [{
+            'group': group_id,
+            'records_in_blob': records_in_blob,
+            'periodic_timeout': 1,
+        }]
+    }]
+
+    # set datasort_dir only if it is provided
+    if datasort_dir is not None:
+        config[0]['backends'][0]['datasort_dir'] = datasort_dir
+
+    server = Servers(servers=config, path=path)
+
+    try:
+        yield server
+    finally:
+        server.stop()
+
+
+def execute_defrag(session, remote, backend_id, chunks_dir, compact):
+    """Run and wait defrag and check its status via monitor stats
+
+    Args:
+        session(elliptics.Session or elliptics.newapi.Session): session connected to the node
+        remote(str): address of the node
+        backend_id(int): id of the backend where defrag should be triggered
+        chunks_dir(str or None): optional chunks_dir where defrag should store temporary chunks
+        compact(bool): whether compact or defrag should be triggered
+    """
+    remote = elliptics.Address.from_host_port_family(remote)
+
+    start = int(time.time())
+
+    # choose defrag method
+    defrag_method = session.start_compact if compact else session.start_defrag
+
+    if chunks_dir is None:
+        # use short call if chunks_dir isn't specified
+        results = defrag_method(remote, backend_id)
+    else:
+        results = defrag_method(remote, backend_id, chunks_dir)
+
+    # there should be the only result, so get it
+    result = results.get()[0]
+
+    # choose which defrag state should be set by backend
+    defrag_state = elliptics.defrag_state.compact if compact else elliptics.defrag_state.data_sort
+
+    # there is the only backend
+    assert len(result.backends) == 1
+    status = result.backends[0]
+    # check its backend_id
+    assert status.backend_id == backend_id
+    # check its defrag state
+    assert status.defrag_state == defrag_state
+
+    # wait while defrag_state isn't reset
+    while status.defrag_state == defrag_state:
+        result = session.request_backends_status(remote).get()[0]
+        assert len(result.backends) == 1
+        status = result.backends[0]
+        assert status.backend_id == backend_id
+
+    end = int(time.time())
+
+    # sleep 1 second while backend updates cached statistics
+    time.sleep(1)
+
+    result = session.monitor_stat(remote,
+                                  categories=elliptics.monitor_stat_categories.backend,
+                                  backends=[backend_id]).get()[0]
+
+    # fetch backend's global statistics which contains info about last defrag
+    global_stat = result.statistics['backends']['0']['backend']['global_stats']
+    # check start/end time from statistics
+    assert start <= global_stat['datasort_start_time'] <= global_stat['datasort_completion_time'] <= end
```

Comment:
Can we really use such a comprehension, how is it converted to Posish notation, what result does each `<=` operator produce?